### PR TITLE
[Chore](pipeline) add some profile log when pipeline canceled

### PR DESCRIPTION
**Pull request by BiteTheDDDDt**
## Proposed changes

try to fix probabilistic hang on union operator when enable pipeline engine

## Further comments

If this is a relatively large or complex change, kick off the discussion at [dev@doris.apache.org](mailto:dev@doris.apache.org) by explaining why you chose the solution you did and what alternatives you considered, etc...

### CODE REVIEW

**Comment by BiteTheDDDDt** [comment]
run buildall

**Comment by BiteTheDDDDt** [comment]
run buildall

**Comment by BiteTheDDDDt** [comment]
run buildall

**Comment by BiteTheDDDDt** [comment]
run buildall

File: be/src/pipeline/pipeline_task.cpp [hunk 1]
```diff
@@ -64,6 +65,8 @@ void PipelineTask::_init_profile() {
     _prepare_timer = ADD_CHILD_TIMER(_task_profile, "PrepareTime", exec_time);
     _open_timer = ADD_CHILD_TIMER(_task_profile, "OpenTime", exec_time);
     _get_block_timer = ADD_CHILD_TIMER(_task_profile, "GetBlockTime", exec_time);
+    _get_block_times_counter = ADD_COUNTER(_task_profile, "GetBlockTimes", TUnit::UNIT);
```
Comment (HappenLee) [inline be/src/pipeline/pipeline_task.cpp:68]: do not name times，‘GetBlockCounter’

File: be/src/pipeline/pipeline_task.cpp [hunk 2]
```diff
@@ -213,7 +216,11 @@ Status PipelineTask::execute(bool* eos) {
         // Pull block from operator chain
         {
             SCOPED_TIMER(_get_block_timer);
+            _get_block_times_counter->update(1);
             RETURN_IF_ERROR(_root->get_block(_state, block, _data_state));
+            if (_block->rows() == 0) {
+                _get_empty_block_times_counter->update(1);
```
Comment (HappenLee) [inline be/src/pipeline/pipeline_task.cpp:222]: why need check empty block ？

**Comment by BiteTheDDDDt** [comment]
run buildall

**Comment by BiteTheDDDDt** [comment]
run buildall

**Review by HappenLee — APPROVED** [review]
LGTM

**Review by zhangstar333 — APPROVED** [review]

**Pull request merged**